src/pratt.rs

```diff
@@ -149,18 +153,33 @@ pub fn right(binding_power: u16) -> Associativity {
     Associativity::Right(binding_power)
 }
 
+/// Specifies a not-associative [`Associativity`] with the given binding power.
+///
+/// Not-associative operators are evaluated from the left-most terms, moving rightward.
+pub fn non(binding_power: u16) -> Associativity {Associativity::Non(binding_power)}
+
 impl Associativity {
     fn left_power(&self) -> u32 {
         match self {
-            Self::Left(x) => *x as u32 * 2,
-            Self::Right(x) => *x as u32 * 2 + 1,
+            Self::Left(x) => *x as u32 * 3,
+            Self::Right(x) => *x as u32 * 3,
+            Self::Non(x) => *x as u32 * 3,
         }
     }
 
     fn right_power(&self) -> u32 {
         match self {
-            Self::Left(x) => *x as u32 * 2 + 1,
-            Self::Right(x) => *x as u32 * 2,
+            Self::Left(x) => *x as u32 * 3 + 1,
+            Self::Right(x) => *x as u32 * 3,
+            Self::Non(x) => *x as u32 * 3 + 1,
+        }
+    }
+
+    fn next_power(&self) -> u32 {
+        match self {
+            Self::Left(x) => *x as u32 * 3,
+            Self::Right(x) => *x as u32 * 3,
+            Self::Non(x) => *x as u32 * 3 - 1,
```

Comment:
For this to be valid, all powers should have 1 added to them such that `*x as u32 * 3 - 1` can never result in underflow.

Reply:
Would this resolve the issue here?
```
            Self::Left(x) => *x as u32 * 3 + 1,
            Self::Right(x) => *x as u32 * 3 + 1,
            Self::Non(x) => *x as u32 * 3 - 1,
```